### DISCUSSION TITLE
How can i get themeConfig

### DISCUSSION THREAD

**Discussion by zRains** [vuepress/core#67] [Q&A] [answered]
i can not get themeConfig from `useSiteData`

**Comment by meteorlxy** [comment, marked as answer]
Use [@vuepress/plugin-theme-data](https://vuepress.github.io/reference/plugin/theme-data.html)

**Reply by yzx9** [reply]
Why don't we directly export composition APIs or symbol, but take this way:

```javascript
import { useThemeData } from '@vuepress/plugin-theme-data/lib/composables'
```

My ideas are:

```javascript
import { useThemeData } from '@vuepress/plugin-theme-data'
```

```javascript
import { inject } from "vue"
import { ThemeDataSymbol } from '@vuepress/plugin-theme-data'

const themeData = inject(ThemeDataSymbol)
```

**Reply by Mister-Hope** [reply]
we are using cjs on node side(plugin and theme entrance), but esm on appSetup, appEnhance or component files. We can not mix them.

So your first thought is impossible. Also we do not expose amy symbols in vue, only use them internally. That's the API design.

**Reply by meteorlxy** [reply]
@Mister-Hope suggested to provide both cjs and esm builds. I think we should not mix them up, but I agree that providing convenience for users is important.

With an alias like `'@vuepress/plugin-theme-data': '@vuepress/plugin-theme-data/lib/composables'` in the plugin could help, but the types definition should be modified, too.

Feel free to discuss if you have any ideas.

**Reply by yzx9** [reply]
@meteorlxy  @Mister-Hope Thanks for you detail reply :-)
